aea/cli/check_packages.py

```diff
@@ -258,6 +309,30 @@ def check_author(configuration_file: Path, expected_author: str) -> None:
         raise UnexpectedAuthorError(configuration_file, expected_author, actual_author)
 
 
+def check_public_id(configuration_file: Path) -> None:
+    """Check the public_id in the code and configuration match."""
+    expected_public_id = get_public_id_from_yaml(configuration_file)
+    # remove last 's' character (as package type is plural in packages directory)
+    package_type_str = configuration_file.parent.parent.name[:-1]
+    package_type = PackageType(package_type_str)
+    if package_type == PackageType.CONNECTION:
+        module_name_to_load = Path("connection.py")
+    elif package_type == PackageType.SKILL:
+        module_name_to_load = Path("__init__.py")
+    else:
+        # no check to do.
+        return
+    module_path_to_load = configuration_file.parent / module_name_to_load
+    content = module_path_to_load.read_text()
+    matches = re.findall("^PUBLIC_ID = (.*)", content, re.MULTILINE)
```

Comment:
Tiny improvement: allow zero or more spaces.
```suggestion
    matches = re.findall("^PUBLIC_ID *= *(.*)", content, re.MULTILINE)
```